src/react/11-state.md

```diff
@@ -0,0 +1,270 @@
+            {/* Use heldText rather than displayText */}
+            <input type="text" value={heldText} onChange={onSetText}/>
+        </div>
+    )
+}
+```
+
+Next, let's wire up that button. We'll define a `onButtonClick` callback to handle changes to the state variables. Pass the callback down to `ButtonComponent`.
+
+```
+function WrapperContainer() {
+    const [ displayText, setDisplayText ] = useState('')
+    const [ heldText, setHeldText ] = useState('')
+
+    function onButtonClick() {
+        // save the new displayText
+        setDisplayText(heldText)
+        // reset the text input
+        setHeldText('')
+    }
+
+    return (
+        <div>
+            <div className="layout-left">
+                <DisplayComponent 
+                    heldText={heldText} 
+                    setHeldText={setHeldText}
+                    displayText={displayText}/>
+            </div>
+            <div className="layout-right">
+                <ButtonComponent onButtonClick={onButtonClick}/>
+            </div>
+        </div>
+    )
+}
+```
+
+Lastly, update the `ButtonComponent` to use our new callback.
+
+```
+function ButtonComponent(props) {
+    const { onButtonClick } = props
+    return (
+        <button type="button" onClick={onButtonClick}>
+            Set Message
+        </button>
+    )
+}
+```
+
+
+### Completed Example ###
+```
+import React, { useState } from 'react'
+
+function WrapperContainer() {
+    // define the displayText to be shown
+    const [ displayText, setDisplayText ] = useState('')
+    // create a holder for unsaved text
+    const [ heldText, setHeldText ] = useState('')
+
+    function onButtonClick() {
+        // save the new displayText
+        setDisplayText(heldText)
+        // reset the text input
+        setHeldText('')
+    }
+
+    return (
+        <div>
+            <div className="layout-left">
+                <DisplayComponent 
+                    heldText={heldText} 
+                    setHeldText={setHeldText}
+                    displayText={displayText}/>
+            </div>
+            <div className="layout-right">
+                <ButtonComponent onButtonClick={onButtonClick}/>
+            </div>
+        </div>
+    )
+}
+
+function DisplayComponent(props) {
+    const { heldText, setHeldText, displayText } = props
+
+    function onSetText(event) {
+        //notice we use the passed in setter defined in the WrapperContainer
+        setHeldText(event.target.value)
+    }
+
+    return (
+        <div>
+            <h1>You entered: {displayText}</h1>
+            {/* Use heldText rather than displayText */}
+            <input type="text" value={heldText} onChange={onSetText}/>
+        </div>
+    )
+}
+
+function ButtonComponent(props) {
+    const { onButtonClick } = props
+    return (
+        <button type="button" onClick={onButtonClick}>
+            Set Message
+        </button>
+    )
+}
+
+
+```
+
+## What data should be kept in React state? ##
```

Comment:
If introduced among the first section, it would be too early to introduce React philosophies. I'd be all for some sort of appendix with "further readings" which could contain these things.